File: 백준/Gold/1338. 알 수 없는 번호/알 수 없는 번호.py
A,B = map(int,input().split())
 
 
if A>B:
    A,B = B,A
 
X,Y = map(int,input().split())
 
X = abs(X)
left_div = A//X
right_div = (B+X)//X
result = []
flag = True
if Y>=abs(X) or Y<0:
    flag = False
if flag:
    for div in range(left_div,right_div+1):
        temp = div*X+Y
        if A<=temp<=B:
            if result:
                flag = False
                break
            else:
                result.append(temp)
 
 
if flag and result:
    print(*result)
else:
    print('Unknwon Number')

File: 백준/Gold/1338. 알 수 없는 번호/알 수 없는 번호.java
import java.io.*;
import java.util.StringTokenizer;

public class Main {

    private static long A, B, X, Y;

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));
        StringTokenizer st = new StringTokenizer(br.readLine());

        A = Long.parseLong(st.nextToken());
        B = Long.parseLong(st.nextToken());

        if (A > B) {
            long temp = A;
            A = B;
            B = temp;
        }

        st = new StringTokenizer(br.readLine());

        X = Math.abs(Long.parseLong(st.nextToken()));
        Y = Long.parseLong(st.nextToken());

        if (0 <= Y && Y < X) {
            long start = Math.floorDiv(A - Y, X) * X + Y;
            if (start < A) {
                start += X;
            }

            if (A <= start && start <= B) {
                if (A <= start + X && start + X <= B) {
                    bw.write("Unknwon Number\n");
                } else {
                    bw.write(String.valueOf(start) + '\n');
                }
            } else {
                bw.write("Unknwon Number\n");
            }
        } else {
            bw.write("Unknwon Number\n");
        }

        br.close();
        bw.close();
    }
}